example-ideal.txt

```diff
@@ -0,0 +1,42 @@
+start toTopLeft
+
+task toTopLeft:
+	start goingLeft
+
+	goingLeft:
+		if isEmpty(Left)
+			move(Left)
+
+		if isWall(Left)
+			next(goingUp)
+
+	goingUp:
+		if isEmpty(Up)
+			move(Up)
+
+		if isWall(Up) and isEmpty(Down)
+			move(Down)
+			next(zigZag)
+
+
+task zigZag:
+	start goingDown
+
+	goingDown:
+		if isEmpty(Down)
+			move(Down)
+
+		if isWall(Down) and isEmpty(Right)
+			move(Right)
+			next(goingUp)
+
+	goingUp:
+		if isEmpty(Up)
+			move(Up)
+
+		if isWall(Up) and isEmpty(Right)
+			move(Right)
+			next(goingDown)
+	
+
+	
```

Comment:
This looks like a pretty flexible ideal syntax. I like that users have a lot of options, though I worry they might be overwhelmed by all of them. Is it really necessary to have tasks as well as rules within the tasks? It seems like you don't need to specify something like next (goingUp) within the zigZag task if it's within the same task and would automatically be run next.